### PR TITLE
find out the setter methods in autovalue builder and add @This

### PR DESCRIPTION
**Pull request by ranmanli**
Will report error "override.return.invalid", since we require explicitly write @This in both the super and child class.  
Screen Shot 2019-06-19 at 12 14 23 AM

### CODE REVIEW

File: bin/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/main/
```
Comment (msridhar) [inline bin/.gitignore]: Why this new `.gitignore`?  Shouldn't we just have a top-level one?

File: src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java [hunk 2]
```diff
@@ -29,6 +43,12 @@ protected TypeAnnotator createTypeAnnotator() {
         return new ListTypeAnnotator(
                 super.createTypeAnnotator(), new ReturnsRcvrTypeAnnotator(this));
     }
+    
+    @Override
+    public TreeAnnotator createTreeAnnotator() {
```
Comment (msridhar) [inline src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java]: We don't need our own TreeAnnotator.  You can just put the `hasAnnotation` method at the top level in this class.

File: src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java [hunk 3]
```diff
@@ -46,7 +67,59 @@ public Void visitExecutable(AnnotatedTypeMirror.AnnotatedExecutableType t, Void
...
+            }
+            
+            boolean inAutoValueBuilder = hasAnnotation(enclosingClass, AutoValue.Builder.class);
+
+            if (inAutoValueBuilder) {
+            	boolean isSetter = methodTree.getName().toString().matches("set.*");
+            	return isSetter;
+            }
+            
+            return inAutoValueBuilder;
```
Comment (msridhar) [inline src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java]: Shouldn't you return false here?

File: src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java [hunk 4]
```diff
@@ -46,7 +67,59 @@ public Void visitExecutable(AnnotatedTypeMirror.AnnotatedExecutableType t, Void
...
+          }
+        
     }
+    
+    private class ReturnsRcvrTreeAnnotator extends TreeAnnotator {
```
Comment (msridhar) [inline src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java]: Move `hasAnnotation` and delete the rest of this inner class

**Review by msridhar — CHANGES_REQUESTED** [review]
Needs a few changes

**Comment by ranmanli** [comment]
Removed the tree annotator

**Comment by msridhar** [comment]
Looks good!

**Pull request merged**